src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.TryCloneFile.OSX.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        private static bool TryCloneFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // Try to clone the file immediately, this will only succeed if the
+            // destination doesn't exist, so we don't worry about locking for this one.
+            int flags = Interop.@libc.CLONE_ACL;
+            retry1:
+            if (Interop.@libc.clonefile(sourceFullPath, destFullPath, flags) == 0)
+            {
+                // Success.
+                return true;
+            }
+            Interop.Error error = Interop.Sys.GetLastError();
+
+            // Some filesystems don't support ACLs, so may fail due to trying to copy ACLs.
+            // This will disable them and allow trying again (a maximum of 1 time).
+            if (flags != 0 && error == Interop.Error.EINVAL)
```

Comment:
```suggestion
                if (error == Interop.Error.EINVAL && flags != 0)
```
nit, easier to read imo